Formula/readline.rb

```diff
@@ -93,8 +72,11 @@ def install
         return 0;
       }
     EOS
-    system ENV.cc, "-L", lib, "test.c", "-L#{lib}", "-lreadline", "-o", "test"
-    assert_equal "test> Hello, World!\nHello, World!",
-      pipe_output("./test", "Hello, World!\n").strip
+
+    ncurses_flags = []
+    ncurses_flags += ["-L#{Formula["ncurses"].opt_lib}", "-lncurses"] if OS.linux?
+
+    system ENV.cc, "-L", lib, "test.c", "-L#{lib}", "-lreadline", *ncurses_flags, "-o", "test"
```

Comment:
Ok, hopefully this won't break too many things.

Reply:
Not sure if these failures are directly related:
```
symbol lookup error: /home/linuxbrew/.linuxbrew/opt/readline/lib/libreadline.so.8: undefined symbol: UP
```
but they appear to happen because that symbol comes from `libtinfo` from `ncurses` and not `readline`: https://stackoverflow.com/questions/46881581/libreadline-so-7-undefined-symbol-up.

Reply:
They are directly related. The correct thing to do would have been to link all these dependencies to `ncurses` directly instead of making them link indirectly via `readline`. This is what Debian do. But, depending on how many formulae will need to be rebuilt to do this, we may just continue with the `SHLIB_LIBS` hack.

It would be nice to be able to drop it, though. `readline` is agnostic about the terminfo library, so users should be able to use it with whichever one they prefer. Linking `readline` directly with `ncurses` breaks this.

Reply:
Does simply making them depend on `ncurses` directly fix the issue?  If so then it seems like a doable fix because the number of affected formulae is quite small and could be fixed in follow up PRs.